flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -316,6 +316,10 @@ public void checkpointState(
             // broadcast cancel checkpoint marker to avoid downstream back-pressure due to
             // checkpoint barrier align.
             operatorChain.broadcastEvent(new CancelCheckpointMarker(metadata.getCheckpointId()));
+            channelStateWriter.abort(
+                    metadata.getCheckpointId(),
+                    new CancellationException("checkpoint aborted via notification"),
+                    true);
```

Comment:
I think you can not relay on this code to be executed. `checkAndClearAbortedStatus` is using best effort `abortedCheckpointIds` set, which can be for example pruned if it grows too large. So if FLINK-26803 doesn't work without this fix, we have to deal with it differently.

Reply:
Thanks for your clarification and reminder, this is really a bug. 

I have updated this PR that fixed two other places where remove aborted checkpoint.


> What exact issue was this bug causing in [FLINK-26803](https://issues.apache.org/jira/browse/FLINK-26803) and what were the symptoms?

The bug will affect the new feature [FLINK-26803](https://issues.apache.org/jira/browse/FLINK-26803), because the channel state file can be closed only after the Checkpoints of all tasks of the shared file are complete or abort. So when the checkpoint of some tasks fails, if abort is not called, the file cannot be closed and all tasks sharing the file cannot execute `inputChannelStateHandles.completeExceptionally(e);` and `resultSubpartitionStateHandles.completeExceptionally(e);` , these AsyncCheckpointRunnable of shared tasks will wait forever.

If the channel state file is not shared between tasks, the current task will not create an AsyncCheckpointRunnable, so the current task and other tasks will not be affected.